File: mobile/modules/kingu-mobile-web-shell/ios/KinguMobileWebShellModule.swift
import ExpoModulesCore

public class KinguMobileWebShellModule: Module {
  public func definition() -> ModuleDefinition {
    Name("KinguMobileWebShell")

    View(KinguMobileWebShellView.self) {
      Events("onLoadState", "onBridgeMessage")

      Prop("generationDirectory") { (view: KinguMobileWebShellView, value: String) in
        view.setGenerationDirectory(value)
      }

      Prop("sessionId") { (view: KinguMobileWebShellView, value: String) in
        view.setSessionId(value)
      }

      Prop("bridgeEnabled") { (view: KinguMobileWebShellView, value: Bool) in
        view.setBridgeEnabled(value)
      }

      AsyncFunction("postBridgeMessage") {
        (view: KinguMobileWebShellView, json: String, promise: Promise) in
        try view.postBridgeMessage(json, promise: promise)
      }

      OnViewDidUpdateProps { (view: KinguMobileWebShellView) in
        view.propsDidUpdate()
      }
    }
  }
}

File: mobile/modules/kingu-mobile-web-shell/android/src/main/java/expo/modules/kingumobilewebshell/KinguMobileWebShellModule.kt
package expo.modules.kingumobilewebshell

import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class KinguMobileWebShellModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("KinguMobileWebShell")

    View(KinguMobileWebShellView::class) {
      Events("onLoadState", "onBridgeMessage", "onExternalNavigation")

      Prop("generationDirectory") { view: KinguMobileWebShellView, value: String ->
        view.setGenerationDirectory(value)
      }

      Prop("sessionId") { view: KinguMobileWebShellView, value: String ->
        view.setSessionId(value)
      }

      Prop("bridgeEnabled") { view: KinguMobileWebShellView, value: Boolean ->
        view.setBridgeEnabled(value)
      }

      AsyncFunction("postBridgeMessage") { view: KinguMobileWebShellView, json: String ->
        view.postBridgeMessage(json)
      }

      OnViewDidUpdateProps { view: KinguMobileWebShellView ->
        view.propsDidUpdate()
      }

      OnViewDestroys { view: KinguMobileWebShellView ->
        view.destroyWebView()
      }
    }
  }
}
